cpp/src/arrow/adapters/orc/adapter_test.cc

```diff
@@ -239,7 +241,9 @@ void AssertTableWriteReadEqual(const std::shared_ptr<Table>& input_table,
   write_options.row_index_stride = 5000;
   EXPECT_OK_AND_ASSIGN(auto writer, adapters::orc::ORCFileWriter::Open(
                                         buffer_output_stream.get(), write_options));
-  ARROW_EXPECT_OK(writer->Write(*input_table));
+  for (int i_write = 0; i_write < nb_writes; i_write++) {
```

Comment:
This is writing N times the same table. Can we instead write a vector of different tables?

Reply:
Yes I'll try yo do this.